main.tf

```diff
@@ -264,7 +264,6 @@ resource "aws_codepipeline" "source_build" {
       output_artifacts = ["code"]
 
       configuration {
-        OAuthToken           = "${var.github_oauth_token}"
```

Comment:
isn't the token required for private GitHub repos?

here it says it's always required https://docs.aws.amazon.com/codepipeline/latest/userguide/reference-pipeline-structure.html#action-requirements

we are not working with private repos at all?

Reply:
Can we do something like `ignore_changes` on this attribute?

Reply:
Terraform considers an empty string as always changing, so maybe something like this

```hcl
variable "github_oauth_token" {
  default     = "defaul"
  description = "GitHub Oauth Token with permissions to access private repositories"
}
```

will work for public repos, and we'll have to provide a correct token for private repos

Reply:
I am using a private repo which still works after removing that line, but I might not have fully tested/understood.  

My experience was that I set up the module with a private repo and it worked, but it requested the token change every apply.  However, looking inside the state file there was no token listed in the state.  That is why it always detects changes is that the token does not persist to tfstate for some reason.  I had assumed that it was because it wasn't actually used in *resource "aws_codepipeline" "source_build"*; but rather in *module "build"*.

Reply:
This var required to support trigger build on github commit

Reply:
found a way to ignore changes for the OAuthToken

```hcl
  lifecycle {
    ignore_changes = [
      "stage.0.action.0.configuration.OAuthToken",
      "stage.0.action.0.configuration.%",
    ]
  }
```

https://stackoverflow.com/questions/48243968/terraform-ignore-changes-and-sub-blocks

It should ignore changes, but should we do it?
If you change the token, you won't be able to re-apply anymore because Terraform will not respect the change.
On the other hand, how often would you change the OAuth token?

@osterman  @goruha  what do you think?